dbt/adapters/databricks/connections.py

```diff
@@ -475,6 +477,14 @@ class DatabricksConnectionManager(SparkConnectionManager):
     TYPE: str = "databricks"
     credentials_provider: Optional[TCredentialProvider] = None
 
+    def cancel_open(self) -> List[str]:
+        for run_id in BaseDatabricksHelper.run_ids:
+            logger.debug(f"Cancel python model job: {run_id}")
+            BaseDatabricksHelper.cancel_run_id(run_id, self.credentials_provider.as_dict()['token'], self.credentials_provider.as_dict()['host'])
```

Comment:
ah, this is where you need to retrieve, and here you don't have an instance...we can use singleton pattern maybe?

Reply:
give me an hour to take a crack at refactoring this; I have an idea :)

Reply:
Singleton maybe its a way, Let me try some code.

Reply:
yeah! thank you very much!